packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -1427,5 +1427,56 @@ async function foo() {
         },
       ],
     },
+    {
+      code: `
+const maybeCallable: string | (() => {});
+const definitelyCallable: () => {};
+Promise.resolve().then(() => {}, undefined);
+Promise.resolve().then(() => {}, null);
+Promise.resolve().then(() => {}, 3);
+Promise.resolve().then(() => {}, maybeCallable);
+Promise.resolve().then(() => {}, definitelyCallable);
+
+Promise.resolve().catch(undefined);
+Promise.resolve().catch(null);
+Promise.resolve().catch(3);
+Promise.resolve().catch(maybeCallable);
+Promise.resolve().catch(definitelyCallable);
+      `,
+      errors: [
+        {
+          line: 4,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+        {
+          line: 5,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+        {
+          line: 6,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+        {
+          line: 7,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+        {
+          line: 10,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+        {
+          line: 11,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+        {
+          line: 12,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+        {
+          line: 13,
+          messageId: 'floatingUselessRejectionHandlerVoid',
+        },
+      ],
+    },
```

Comment:
[Non-actionable] Heh, this reminded me to file #6887. No need to make any change here, as this is following the existing standard in the file.